CDP4Addin/AddinPluginLoader.cs

```diff
@@ -34,19 +39,22 @@ public AddinPluginLoader(IAppSettingsService<AddinAppSettings> appSettingsServic
             this.DirectoryCatalogues = new List<DirectoryCatalog>();
 
             var currentPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
-
             var path = Path.Combine(currentPath, PluginDirectoryName);
             var directoryInfo = new DirectoryInfo(path);
+            this.NewPlugins = new List<string>();
 
             if (directoryInfo.Exists)
             {
                 foreach (var dir in directoryInfo.EnumerateDirectories())
                 {
-                    var fileName = Path.GetFileName(dir.FullName);
+                    var pluginName = dir.FullName.ToUpper();
+                    var locationInfo = new DirectoryInfo(pluginName);
+                    var pluginSettingsList = appSettingsService.AppSettings.Plugins.Where(x => (x.Key != null) && (x.Key.ToUpper() == locationInfo.Name)).ToList();
```

Comment:
Here instead of `Where` you can use `FirstOrDefault` because you only care if it is there or not. If it is there more than once you just igonre it, and if it is not there you get  `null`.

```
var existingPlugin = appSettingsService.AppSettings.Plugins.FirstOrDefault(x => (x.Key != null) && (x.Key.ToUpper() == locationInfo.Name));

var isNewPlugin = existingPlugin == null;
```

Reply:
Code has been refactored